test/cli/shared/errors.dart

```diff
@@ -72,6 +72,58 @@ void sharedTests(Future<TestProcess> runSass(Iterable<String> arguments)) {
     await sass.shouldExit(65);
   });
 
+  test("from an error encountered within a function", () async {
+    await d.file("test.scss", """
+@function a() {
+  @error "Within A.";
+}
+
+.b {
+  c: a();
+}
+""").create();
+
+    var sass = await runSass(["--no-unicode", "test.scss"]);
+    expect(
+        sass.stderr,
+        emitsInOrder([
+          "Error: \"Within A.\"",
+          "  ,",
+          "6 |   c: a();",
+          "  |      ^^^",
+          "  '",
+          "  test.scss 2:3  a()",
```

Comment:
I know this is what my example in the original issue looked like, but looking at it closer I think it's confusing that the top line of the stack trace is pointing to a different place than the span. Can you make it regenerate the stack trace at the same time it regenerates the span itself?